### PR TITLE
[DOC] Improve handling of URLs in rendered docs

### LINKED ISSUE #3303: Broken links due to cleaning of `nilearn.github.io`
I noticed some broken links due to URLs referring to pages based on the old website structure. After cleaning `nilearn.github.io` we may find more of these.

### Affected pages
One example is in the docstring of the [`SearchLight` class](https://nilearn.github.io/stable/modules/generated/nilearn.decoding.SearchLight.html#nilearn.decoding.SearchLight) where http://nilearn.github.io/manipulating_images/input_output.html is referenced several times. This should be http://nilearn.github.io/stable/manipulating_images/input_output.html 

I haven't spent much time on this but I assume there are other affected pages.

### LINKED ISSUE #3321: Improve handling and managing of URLs
Working on #3315 made me think that there might be a better way to handle URLs. For example, some links repeat many times making it hard to manage them. I think we should be able to make a mapping of URLs to keywords in one place and then create a sphinx role that would link the keywords. This would make it easier to fix broken links in the future and, if possible to use anchor text, would make the documentation more readable. This is useful for rendered docs and wouldn't necessarily apply to all URLs in the library.

### PR DESCRIPTION
**Pull request by ymzayek**
Closes #3321 and closes #3303.

Changes proposed in this pull request:

- Replace rendered URLs with sphinx internal referencing
- If external referencing is needed, use intersphinx mapping or external links (in `doc/conf.py`)

### CODE REVIEW

**Comment by ymzayek** [comment]
Old rendering:
Screenshot from 2022-08-09 15-55-23

New rendering as implemented in the commited changes:
Screenshot from 2022-08-09 15-57-03

**Comment by ymzayek** [comment]
New rendering with custom anchor text is also possible:
Screenshot from 2022-08-09 16-00-05

File: AUTHORS.rst [hunk 1]
```diff
@@ -103,16 +103,16 @@ We suggest that you read and cite the paper. Thank you.
 Citing scikit-learn
 -------------------
 
-A huge amount of work goes into `scikit-learn <http://scikit-learn.org/>`_,
+A huge amount of work goes into :sklearn:`scikit-learn <>`,
 upon which nilearn relies heavily.
 Researchers who invest their time in developing and maintaining the package
 deserve recognition with citations.
-In addition, the `Parietal team <https://team.inria.fr/parietal/>`_ needs citations
+In addition, the :parietal:`INRIA Parietal Project Team <>` needs citations
```
Comment (bthirion) [inline AUTHORS.rst:110]: Can you update parietal -> mind ?

File: AUTHORS.rst [hunk 2]
```diff
@@ -8,11 +8,11 @@ Team
 
 This work is made available by a community of people, which
 originated from
-the `INRIA Parietal Project Team <https://team.inria.fr/parietal/>`_
-and the `scikit-learn <http://scikit-learn.org/>`_ but grew much further.
+the :parietal:`INRIA Parietal Project Team <>` and :sklearn:`scikit-learn <>`
```
Comment (bthirion) [inline AUTHORS.rst:11]: Can you update parietal -> mind ?

**Review by bthirion — COMMENTED** [review]
Thx ! This is a big improvement overall.

**Review by NicolasGensollen — APPROVED** [review]
Great ! Thanks @ymzayek ! 👍 
Not sure if this is finished or not, but LGTM at this point !

**Comment by ymzayek** [comment]
Yes I believe this one is ready, but if I missed something let me know

**Review by bthirion — APPROVED** [review]
I'm happy with the changes. thx !

**Comment by ymzayek** [comment]
I will have a final look through the artifacts of the full build and then merge

**Comment by ymzayek** [comment]
Merging

**Pull request merged**